lib/usagereporter/teleport/types.go

```diff
@@ -334,6 +334,29 @@ func (u *SFTPEvent) Anonymize(a utils.Anonymizer) prehogv1.SubmitEventRequest {
 	}
 }
 
+// AgentMetadataEvent is an event emitted after an agent first connects to the auth server.
+type AgentMetadataEvent prehogv1.AgentMetadataEvent
+
+func (u *AgentMetadataEvent) Anonymize(a utils.Anonymizer) prehogv1.SubmitEventRequest {
+	return prehogv1.SubmitEventRequest{
+		Event: &prehogv1.SubmitEventRequest_AgentMetadataEvent{
+			AgentMetadataEvent: &prehogv1.AgentMetadataEvent{
+				Version:               u.Version,
+				HostId:                u.HostId,
```

Comment:
This might not be just a UUID.
```suggestion
				HostId:                a.AnonymizeString(u.HostId),
```

Reply:
Thanks for catching this! 20a7c22321721e5fb989efe826119b70dae2efba